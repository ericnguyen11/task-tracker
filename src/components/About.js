import { Link } from "react-router-dom";

const About = () => {
  return (
    <div data-testid="about">
      <h4>Version 1.0.1</h4>
      <Link className="about-link" to="/">
        Go back
      </Link>
    </div>
  );
};

export default About;
