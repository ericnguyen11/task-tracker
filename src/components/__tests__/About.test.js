import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { render, screen, cleanup } from "@testing-library/react";

import { shallow, mount } from "enzyme";

import About from "../About.js";

// Unmounts React trees from DOM after each test. Unnecessary with enzyme shallow
afterEach(() => {
  cleanup();
});

it("renders about component that includes version", () => {
  render(
    <MemoryRouter>
      <About />
    </MemoryRouter>
  );
  const aboutElement = screen.getByTestId("about");
  expect(aboutElement).toBeInTheDocument();
  expect(screen.getByText("Version 1.0.1")).toBeInTheDocument();
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <About />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

/* ENZYME TESTS */

it("renders without crashing", () => {
  shallow(<About />);
});

it("matches snapshot", () => {
  const wrapper = shallow(<About />);
  expect(wrapper).toMatchSnapshot();
});

it("includes link to home", () => {
  const wrapper = mount(
    <MemoryRouter>
      <About />
    </MemoryRouter>
  );
  expect(wrapper.find(".about-link").at(0).props().to).toBe("/");
});
