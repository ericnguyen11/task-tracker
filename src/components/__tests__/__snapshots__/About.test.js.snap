// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  data-testid="about"
>
  <h4>
    Version 1.0.1
  </h4>
  <a
    className="about-link"
    href="/"
    onClick={[Function]}
  >
    Go back
  </a>
</div>
`;

exports[`matches snapshot 2`] = `
<div
  data-testid="about"
>
  <h4>
    Version 1.0.1
  </h4>
  <Link
    className="about-link"
    to="/"
  >
    Go back
  </Link>
</div>
`;
